# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../liblockfile/glibc24.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./lockfile.c.orig	2006-06-15 15:54:38.000000000 +0200
+++ ./lockfile.c	2006-06-15 15:57:43.000000000 +0200
@@ -54,7 +54,7 @@
 #ifdef LIB
 static
 #endif
-int eaccess(char *fn, gid_t gid, struct stat *st)
+int eaccess2(char *fn, gid_t gid, struct stat *st)
 {
 	struct stat	tmp;
 	uid_t		uid = geteuid();
@@ -96,7 +96,7 @@
 		*p = 0;
 	else
 		strcpy(dir, ".");
-	if (eaccess(dir, egid, NULL) >= 0) {
+	if (eaccess2(dir, egid, NULL) >= 0) {
 		free(dir);
 		return 0;
 	}
@@ -111,7 +111,7 @@
 			return 0;
 		mailgid = st.st_gid;
 	}
-	ret = eaccess(dir, mailgid, NULL) >= 0;
+	ret = eaccess2(dir, mailgid, NULL) >= 0;
 	free (dir);
 	return ret;
 }
--- ./dotlockfile.c.orig	2006-06-15 15:57:21.000000000 +0200
+++ ./dotlockfile.c	2006-06-15 15:58:16.000000000 +0200
@@ -42,7 +42,7 @@
 extern int optind;
 #endif
 
-extern int eaccess(char *, gid_t, struct stat *);
+extern int eaccess2(char *, gid_t, struct stat *);
 
 /*
  *	Sleep for an amout of time while regulary checking if
@@ -252,14 +252,14 @@
 		return L_ERROR;
 	}
 	gid = getgid();
-	if (eaccess(dir, gid, &st) < 0) {
+	if (eaccess2(dir, gid, &st) < 0) {
 		if (errno == ENOENT) {
 enoent:
 			if (!quiet) fprintf(stderr,
 				"dotlockfile: %s: no such directory\n", dir);
 			return L_TMPLOCK;
 		}
-		if ((r = eaccess(dir, getegid(), &st) < 0) && errno == ENOENT)
+		if ((r = eaccess2(dir, getegid(), &st) < 0) && errno == ENOENT)
 			goto enoent;
 		if (r < 0 || !ismaillock(lockfile, pwd->pw_name)) {
 			if (!quiet) fprintf(stderr,
